Extension/src/main.ts

```diff
@@ -75,38 +75,70 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
-        vscode.window.showErrorMessage(errMsg, downloadLink).then(async (selection) => {
-            if (selection === downloadLink) {
-                vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
-            }
-        });
-    } else if (!(await util.checkInstallBinariesExist())) {
-        errMsg = localize("extension.installation.failed", "The C/C++ extension failed to install successfully. You will need to repair or reinstall the extension for C/C++ language features to function properly.");
-        const reload: string = localize("remove.extension", "Attempt to Repair");
-        vscode.window.showErrorMessage(errMsg, reload).then(async (value?: string) => {
-            if (value === reload) {
-                await util.removeInstallLockFile();
-                vscode.commands.executeCommand("workbench.action.reloadWindow");
-            }
-        });
-    } else if (!(await util.checkInstallJsonsExist())) {
-        // Check the Json files to declare if the extension has been installed successfully.
-        errMsg = localize("jason.files.missing", "The C/C++ extension failed to install successfully. You will need to reinstall the extension for C/C++ language features to function properly.");
         const downloadLink: string = localize("download.button", "Go to Download Page");
-        vscode.window.showErrorMessage(errMsg, downloadLink).then(async (selection) => {
-            if (selection === downloadLink) {
-                vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
+        if (installedPlatformAndArchitecture.platform === 'darwin' && installedPlatformAndArchitecture.architecture === "x64" && arch === "arm64") {
+            if (promptForMacArchictureMismatch.Value) {
+                // Display a message specifically referring the user to the ARM64 Mac build on ARM64 Mac.
+                errMsg = localize("native.binaries.mismatch.osx", "This {0} version of the extension has been installed.  Since you are on an Apple Silicon Mac, we recommend installing the \"{1}\" version of the extension.", GetOSName(installedPlatformAndArchitecture.platform), vsixName);
+                promptForMacArchictureMismatch.Value = false;
+                vscode.window.showErrorMessage(errMsg, downloadLink).then(async (selection) => {
+                    if (selection === downloadLink) {
+                        vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
+                    }
+                });
             }
-        });
+        } else {
+            // Reset the persistent boolean tracking whether to warn the user of architecture mismatch on OSX.
+            promptForMacArchictureMismatch.Value = true;
+            errMsg = localize("native.binaries.not.supported", "This {0} version of the extension is incompatible with your OS. Please download and install the \"{1}\" version of the extension.", GetOSName(installedPlatformAndArchitecture.platform), vsixName);
```

Comment:
`GetOSName(installedPlatformAndArchitecture.platform)` also add `GetOSName(installedPlatformAndArchitecture.arch)`. platform will just return "darwin"?

Reply:
I changed this to refer directly to "Intel" and "Apple Silicon", since this message is only for that scenario.

Reply:
What? It returns MacOS, right?